src/EnergyPlus/Pumps.cc

```diff
@@ -281,950 +272,898 @@ void GetPumpInput(EnergyPlusData &state)
+                                 format("{} {} = {}, Invalid {}", RoutineName, cCurrentModuleObject, thisPump.Name, thisInput->cAlphaFieldNames(6)));
+                ShowContinueError(state, format("Schedule named =[{}]. was not found and will not be used.", thisInput->cAlphaArgs(6)));
+            }
+        }
+
+        thisPump.NomVolFlowRate = thisInput->rNumericArgs(1);
+        if (thisPump.NomVolFlowRate == AutoSize) {
+            thisPump.NomVolFlowRateWasAutoSized = true;
+        }
+        thisPump.NumPumpsInBank = thisInput->rNumericArgs(2);
+        thisPump.NomPumpHead = thisInput->rNumericArgs(3);
+        thisPump.NomPowerUse = thisInput->rNumericArgs(4);
+        if (thisPump.NomPowerUse == AutoSize) {
+            thisPump.NomPowerUseWasAutoSized = true;
+        }
+        thisPump.MotorEffic = thisInput->rNumericArgs(5);
+        thisPump.FracMotorLossToFluid = thisInput->rNumericArgs(6);
+        thisPump.PartLoadCoef[0] = 1.0;
+        thisPump.PartLoadCoef[1] = 0.0;
+        thisPump.PartLoadCoef[2] = 0.0;
+        thisPump.PartLoadCoef[3] = 0.0;
+
+        if (!thisInput->lAlphaFieldBlanks(7)) { // zone named for pump skin losses
+            thisPump.ZoneNum = UtilityRoutines::FindItemInList(thisInput->cAlphaArgs(7), state.dataHeatBal->Zone);
+            if (thisPump.ZoneNum > 0) {
+                thisPump.HeatLossesToZone = true;
+                if (!thisInput->lNumericFieldBlanks(7)) {
+                    thisPump.SkinLossRadFraction = thisInput->rNumericArgs(7);
                 }
             } else {
                 ShowSevereError(state,
-                                cCurrentModuleObject + "=\"" + state.dataIPShortCut->cAlphaArgs(1) + "\" invalid " +
-                                    state.dataIPShortCut->cAlphaFieldNames(7) + "=\"" + state.dataIPShortCut->cAlphaArgs(7) + "\" not found.");
+                                format("{}={} invalid {}={} not found.",
+                                       cCurrentModuleObject,
+                                       thisPump.Name,
+                                       thisInput->cAlphaFieldNames(7),
+                                       thisInput->cAlphaArgs(7)));
                 ErrorsFound = true;
             }
         }
-        if (!state.dataIPShortCut->lAlphaFieldBlanks(8)) {
-            if (state.dataIPShortCut->cAlphaArgs(8) == "POWERPERFLOW") {
-                state.dataPumps->PumpEquip(PumpNum).powerSizingMethod = PowerSizingMethod::SizePowerPerFlow;
-            } else if (state.dataIPShortCut->cAlphaArgs(8) == "POWERPERFLOWPERPRESSURE") {
-                state.dataPumps->PumpEquip(PumpNum).powerSizingMethod = PowerSizingMethod::SizePowerPerFlowPerPressure;
-            } else {
+        if (!thisInput->lAlphaFieldBlanks(8)) {
+            thisPump.powerSizingMethod = static_cast<PowerSizingMethod>(
+                getEnumerationValue(powerSizingMethodNamesUC, UtilityRoutines::MakeUPPERCase(state.dataIPShortCut->cAlphaArgs(8))));
```

Comment:
Good idea--I'd ask that we pin that and come back later.  There are lots of places in the code where this is going to be an issue.  Seems like it would be best if someone went through the code and dealt with these all at once.

Reply:
Sure, why not.

Reply:
Yeah, using the template function to static cast it for us is a great idea.